engine/engine_test.go

```diff
@@ -323,3 +338,615 @@ func TestSetDefaultWebsocketDataHandler(t *testing.T) {
 		t.Fatalf("received: '%v' but expected: '%v'", err, nil)
 	}
 }
+
+func TestAllExchangeWrappers(t *testing.T) {
+	t.Parallel()
+	cfg := config.GetConfig()
+	err := cfg.LoadConfig("../testdata/configtest.json", true)
+	if err != nil {
+		t.Fatal("load config error", err)
+	}
+	for i := range cfg.Exchanges {
+		name := cfg.Exchanges[i].Name
+		if common.StringDataContains(unsupportedExchangeNames, strings.ToLower(name)) {
+			continue
+		}
+
+		exch, assetPairs := setupExchange(t, name, cfg)
+		t.Run(name+" wrapper tests", func(t *testing.T) {
+			t.Parallel()
+			executeExchangeWrapperTests(t, exch, assetPairs)
+		})
+	}
+}
+
+func TestAllExchangeWebsockets(t *testing.T) {
+	t.Parallel()
+	t.Skip("testing while PR is open")
```

Comment:
Will do. This is mostly just because this test alerts to a lot of races via `go test ./... -race` in Github Actions. I'll be looking into whats causing it

Reply:
🚒 🧑‍🚒

Reply:
Putting out fires